### ISSUE TITLE
Reranker scores are blended on incompatible scales, breaking retrieval ordering for several providers

### ISSUE THREAD

**Issue by blinkeye-lcm** [infiniflow/ragflow#15428]
### Describe the bug

The hybrid score in `rag/nlp/search.py` (`rerank_by_model`) blends the reranker output with token similarity using a fixed weighting that assumes both terms sit on the same `[0, 1]` scale:

```python
return tkweight * np.array(tksim) + vtweight * vtsim + rank_fea  # tkweight=0.3, vtweight=0.7
```

`vtsim` is the raw output of `rerank_mdl.similarity()`, but the reranker implementations in `rag/llm/rerank_model.py` return scores on very different scales and only three of them normalize:

- `NvidiaRerank` returns raw logits (`d["logit"]`), which are unbounded and frequently negative. With `vtweight=0.7`, a negative logit drags a genuinely relevant chunk below pure keyword matches, and a magnitude such as ±10 swamps `tksim`, which lives in `[0, 1]`. Reranking actively degrades results.
- Most providers (Cohere, Jina, Voyage, SiliconFlow, QWen, and others) return a `relevance_score` that is usually, but not guaranteed, in `[0, 1]`.
- Only `LocalAIRerank`, `OpenAI_APIRerank`, and `RAGconRerank` call the existing   `Base._normalize_rank`.

As a result, the same query produces differently scaled scores depending on the configured reranker, and logit based providers are outright broken.

### Additional context

`rerank_by_model` also contains twelve leftover `print("[DEBUG ...]")` statements introduced in #14231. They run on every retrieval, add per chunk overhead, and leak user queries, keywords, and document content to stdout and container logs.

### Expected behavior

Every reranker should feed the hybrid blend on a single, comparable `[0, 1]` scale regardless of which provider is configured, and no debug output should be emitted on the retrieval path.
